version: '3'

services:
  gunicorn:
    build: 
      context: .
      dockerfile: ./gunicorn_Dockerfile
    image: gunicorn:20.1.0
    container_name: nosta_gunicorn
    depends_on:
      - db
    networks:
      - nosta
    environment:
      - DJANGO_SETTINGS_MODULE=nosta.settings.production
    volumes:
      - .:/usr/src/app/
      - gunicorn:/var/run/gunicorn
    stdin_open: true
    tty: true

  nginx:
    image: nginx:1.17.7
    container_name: nosta_nginx
    depends_on:
      - gunicorn
    networks:
      - prasic
      - nosta
    volumes:
      - ./gunicorn.conf:/etc/nginx/conf.d/default.conf
      - gunicorn:/var/run/gunicorn
      - ./static/assets:/home/static

    labels:
      - traefik.enable=true
      - traefik.http.routers.nosta_nginx.rule=Host(`nosta.${DOMAIN}`)
      - traefik.http.routers.nosta_nginx.entrypoints=https
      - traefik.http.routers.nosta_nginx.tls=true
      - traefik.http.services.nosta_nginx.loadbalancer.server.port=80
      - traefik.http.routers.nosta_nginx.tls.certresolver=${CERT_RESOLVER}
  
  db:
    image: mariadb:10.6.4
    restart: always
    container_name: nosta_db
    networks:
      - nosta
    environment: 
      MARIADB_ROOT_PASSWORD: nosta_db_root_password
    volumes:
      - ./mysql:/var/lib/mysql

volumes:
  gunicorn:
    driver: local

networks:
  prasic:
    external: true
  nosta:
    external: true